nncf/experimental/torch/fx/constant_folding.py

```diff
@@ -252,6 +265,8 @@ def constant_fold(
         for node in erased_params:
             gm.graph.erase_node(node)
 
-        gm.graph.eliminate_dead_code()
+        # Custom _is_impure function allows to eliminate all layers with zero
+        # users including inplace ops like relu_ besides output and placeholders.
+        gm.graph.eliminate_dead_code(_is_impure)
```

Comment:
It looks like this is the default behavior where `node.is_impure()` is checked which checks if node.op is placeholder or output https://github.com/pytorch/pytorch/blob/c418a9ac7501184d4a8cb4443f49e727d43f6be0/torch/fx/graph.py#L1815

Reply:
Oh, alright. I went over the documentation (https://pytorch.org/docs/stable/fx.html#torch.fx.Node.is_impure) instead of the code.